hs_core/hydroshare/resource.py

```diff
@@ -962,7 +962,7 @@ def delete_format_metadata_after_delete_file(resource, file_name):
             resource.metadata.delete_element(format_element.term, format_element.id)
 
 
-def delete_resource_file(pk, filename_or_id, user):
+def delete_resource_file(pk, filename_or_id, user, delete_logical_file=True):
```

Comment:
I don't understand why this option is at this level. It would seem that one would delete the logical file automagically when its last physical file was deleted. This would be a natural cascade.